src/devices/openxrheadset/OpenXrInterface.cpp

```diff
@@ -1704,36 +1832,59 @@ void OpenXrInterface::getAllPoses(std::vector<NamedPoseVelocity> &additionalPose
         }
     }
 
+    numberOfPoses += 52;  // adding also fingers
     additionalPoses.resize(numberOfPoses);
 
     size_t poseIndex = 0;
 
     auto& head = additionalPoses[poseIndex];
     head.name = "openxr_head";
     head.pose = headPose();
-    head.velocity = headVelocity();
+	head.velocity = headVelocity();
+    head.filterType = PoseFilterType::NONE;
     poseIndex++;
 
     auto& left_arm = additionalPoses[poseIndex];
     left_arm.name = "openxr_left_hand";
-    left_arm.pose = leftHandPose();
-    left_arm.velocity = leftHandVelocity();
+    left_arm.pose = m_pimpl->leftHandPose_;
+    if (m_pimpl->hand_tracking_supported)
+        left_arm.filterType = PoseFilterType::NONE;
     poseIndex++;
 
     auto& right_arm = additionalPoses[poseIndex];
     right_arm.name = "openxr_right_hand";
-    right_arm.pose = rightHandPose();
-    right_arm.velocity = rightHandVelocity();
+    right_arm.pose = m_pimpl->rightHandPose_;
+    if (m_pimpl->hand_tracking_supported)
+        right_arm.filterType = PoseFilterType::NONE;
     poseIndex++;
 
-    for (size_t topLevelIndex = 0; topLevelIndex <  m_pimpl->top_level_paths.size(); ++topLevelIndex)
-    {
-        std::vector<PoseAction>& posesList = m_pimpl->top_level_paths[topLevelIndex].currentActions().poses;
-        for (size_t i = (topLevelIndex < 2) ? 1 : 0; i < posesList.size(); ++i) // if we are in the first or the second top level path (hence left or right hand), we start from the index 1 (the first one is the default)
+    // add also all finger poses
+    if (m_pimpl->hand_tracking_supported) {
+        for (size_t i = 0; i < m_pimpl->leftHandJointPoses_.size(); ++i)
         {
-            additionalPoses[poseIndex] = posesList[i];
+            auto& finger = additionalPoses[poseIndex];
+            finger.name = "openxr_left_hand_finger_" + std::to_string(i);
+            finger.pose = m_pimpl->leftHandJointPoses_[i];
+            finger.filterType = PoseFilterType::NONE;
             poseIndex++;
         }
```

Comment:
Could you add the code for the fingers after this ``for``? The order should not matter here, but it would be cleaner